### DISCUSSION TITLE
I made a small and unique Hydrogen Drumkit.

### DISCUSSION THREAD

**Discussion by ghost** [hydrogen-music/hydrogen#1700] [Show and tell]
![BJDNielSounds](https://yggdrasil-studio.github.io/BJDNielSounds/images/BJDNielSounds.png)

I made a Percussions drumkit called `BJDNielPercussions`. It is my hobby sound collection project created with Ball-Jointed Doll **Niel**.

Please check out it: https://yggdrasil-studio.github.io/BJDNielSounds/

Happy New Year!

**Comment by theGreatWhiteShark** [comment]
Thanks for sharing!

How did you created the .h2drumkit? From its looks it uses a quite ancient format and fails to load at least on my machine.

**Reply by ghost** [reply]
I test it on **Hydrogen 1.0.1 [Feb 19 2021]** because I'm now using Raspberry Pi only.

`h2drumkit` that I made with `tar` archive. Its `drumkit.xml` is [here](https://github.com/YGGDRASIL-STUDIO/BJDNielSounds/blob/main/BJDNielPercussions/BJDNielPercussions/drumkit.xml). Any suggestions for newer version?

**Reply by ghost** [reply]
I found that I made with `czvf`, so I replaced `tar` archive method to `cvf`. Can you test with [new file](https://github.com/YGGDRASIL-STUDIO/BJDNielSounds/releases/download/v1.0.0/BJDNielPercussions.tar.gz)?

**Reply by theGreatWhiteShark** [reply]
> I test it on Hydrogen 1.0.1

Hmm. If it was able to load with this version but fails with the current on, than that's a regression. I'll check.

> I found that I made with czvf, so I replaced tar archive method to cvf

It's not about the bundling. The `drumkit.xml` you create uses a quite old format. Since almost 10 years both the drumkit and the individual instruments have 'components'. How did you created this particular file?

The default way to created it is via the GUI Using Drumkit > New in the menu bar and then adding the individual instruments and their layers/samples. If you are more of a scripting type, you can use one of our [default kits](https://github.com/hydrogen-music/hydrogen/tree/master/data/drumkits) as a template, fill in your samples and meta data, and discard everything else.

Problem with this old format is that it will only be loaded by some legacy code and is not guaranteed to work properly. (Some additional parameters have been introduce along the way and others have been dropped).

**Reply by ghost** [reply]
Thank you for your advice. I uploaded the new [drumkit.xml](https://github.com/YGGDRASIL-STUDIO/BJDNielSounds/blob/main/BJDNielPercussions/BJDNielPercussions/drumkit.xml) and [h2drumkit](https://github.com/YGGDRASIL-STUDIO/BJDNielSounds/releases/download/v1.0.0/BJDNielPercussions.tar.gz)!

**Reply by theGreatWhiteShark** [reply]
Awesome. Works like a charm.

Small note: it still does not use the most recent format as `pan_L` and `pan_R` have been fused to `pan` one or two years ago. You can also check the validity of the `drumkit.xml` file using the command.

```bash
xmllint --schema PATH_TO_HYDROGEN_REPO/data/xsd/drumkit.xsd drumkit.xml
```

**Reply by ghost** [reply]
Thanks great. `pan` doesn't seem to work in the old version like me. Can I use both of `pan_LR` with `pan`? `xsd` says `choice`, and there are no `drumkit.xml validates` message if I use both.

**Reply by theGreatWhiteShark** [reply]
Ah, sorry. I got confused. The XML linter did not complain about the `pan` property but the one(s) next to it. Both `isSoloed` and `pitchOffset` are missing in your instrument definitions. They were introduced about two years ago. 

Did you use the 1.1.1 release to create the kit? with the 1.2.0-beta one it should work right away

**Reply by ghost** [reply]
I've tested with `isSoloed` and `pitchOffset` after your previous reply.

But the `pan` doesn't work on my machine, `Hydrogen 1.0.1 from feb 2021`, installed from APT.

**Reply by ghost** [reply]
Now I'm on recent Hydrogen version built from git,

I use both of old `pan_LR` with recent `pan` tags, it working on both of old and recent versions!

**Reply by theGreatWhiteShark** [reply]
> I use both of old pan_LR with recent pan tags, it working on both of old and recent versions!

I think the proper way for now is to use the older version with `pan_L` and `pan_R` and omitting `pan`. Else older versions of Hydrogen won't be able to properly load the kit. In 10 years when all the older versions have been replaced, we will switch to `pan` as the default choice.

**Reply by ghost** [reply]
Then I will set in `pan_L` and `pan_R` for now.
